### PR TITLE
(maint) Prepare for facter integration

### PR DESCRIPTION
**Pull request by caseywilliams**
This commit cleans up cmake configs and moves a few files around in
preparation for libwhereami's use in facter (see FACT-1742).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,12 +27,19 @@ include(options)
...
-add_definitions(-DLEATHERMAN_I18N)
-## Pull in helper macros for working with leatherman libraries
-include(leatherman)
+
+if(LEATHERMAN_USE_LOCALES)
+    add_definitions(-DLEATHERMAN_I18N)
+    set(BOOST_COMPONENTS locale)
+endif()
+
+list(APPEND BOOST_COMPONENTS filesystem regex)
```
Comment (branan) [inline CMakeLists.txt:36]: Were we just using these accidentally without specifying them before?

File: lib/inc/whereami/result.hpp [hunk 2]
```diff
@@ -69,7 +75,7 @@ namespace whereami { namespace detectors {
         /**
          * Metadata about the hypervisor
          */
-        metadata metadata_;
+        class metadata metadata_;
```
Comment (branan) [inline lib/inc/whereami/result.hpp:78]: This `class` is needed to disambiguate the `metadata` method above, right?

**Review by branan — APPROVED** [review]
All looks sane, unless I get surprising answers to my questions :+1:

File: CMakeLists.txt (hunk 1 above)
Comment (caseywilliams) [reply CMakeLists.txt:36]: Yes! I hadn't even looked at this again before yesterday...

File: lib/inc/whereami/result.hpp (hunk 2 above)
Comment (caseywilliams) [reply lib/inc/whereami/result.hpp:78]: That's right

**Review by mcdonaldseanp — APPROVED** [review]
Nothing seems out of place. 👍

**Pull request merged**